pkg/registrar/map_functions.go

```diff
@@ -127,17 +129,45 @@ func (mapper *Mapper) ClusterSupplyChainToWorkloadRequests(object client.Object)
 }
 
 func (mapper *Mapper) clusterSupplyChainToWorkloads(sc v1alpha1.ClusterSupplyChain) ([]v1alpha1.Workload, error) {
-	list := &v1alpha1.WorkloadList{}
+	err := mapper.addGVK(&sc)
+	if err != nil {
+		mapper.Logger.Error(err, fmt.Sprintf("could not get GVK for supply chain: %s", sc.Name))
+		return nil, err
+	}
+
+	scList := &v1alpha1.ClusterSupplyChainList{}
+	err = mapper.Client.List(context.TODO(), scList)
+	if err != nil {
+		mapper.Logger.Error(err, "cluster supply chain to workloads: client list supply chains")
+		return nil, err
+	}
+
+	var selectorGetters []repository.SelectorGetter
+	for _, item := range scList.Items {
+		item := item
```

Comment:
Is this just to create a copy of the object? Would `persistentItem := item.DeepCopy()` suit? Would it be more grokkable?

Reply:
I think that would work too but I was trying to be consistent with the way the controller finds the best matching SC (see [here](https://github.com/vmware-tanzu/cartographer/blob/main/pkg/repository/repository.go#L286-L290))